content/posts/a-guide-on-commit-messages.mdx

```diff
@@ -0,0 +1,169 @@
+
+```
+fix #72 unnecessary object copies
+Fixing unshift documentation examples
+seq.chain path parameter fix #18
+Remove built files
+Add tests
+Add Circle CI base config
+Rename es -> src
+Add lang package with toggle, fix exports
+Core package
+```
+
+In these commit message we can see a few mistakes in term of style, content and metadata. Chris beams describe well these three points in his article [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
+
+> Style. Markup syntax, wrap margins, grammar, capitalization, punctuation. Spell these things out, remove the guesswork, and make it all as simple as possible. The end result will be a remarkably consistent log that’s not only a pleasure to read but that actually does get read on a regular basis.
+> 
+> Content. What kind of information should the body of the commit message (if any) contain? What should it not contain?
+> 
+> Metadata. How should issue tracking IDs, pull request numbers, etc. be referenced?
+> 
+> Chris Beams - [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+We can see sometime a capitalized message. There are metadata for only two commits. Finally some messages content are so light we can't figure out what the commit does. In the same article Chris goes through 7 rules to write a good commit message.
+
+> 1. Separate subject from body with a blank line
+> 2. Limit the subject line to 50 characters
+> 3. Capitalize the subject line
+> 4. Do not end the subject line with a period
+> 5. Use the imperative mood in the subject line
+> 6. Wrap the body at 72 characters
+> 7. Use the body to explain what and why vs. how
+>
+> Chris Beams - [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+I personnally apply a subset of these rules:
+
+- Limit the subject line to 50 characters
+- Capitalize the subject line
+- Use the imperative mood in the subject line
+
+The character limit does not only force you to be concise which is a great skill but also fits well with a lot of tools as explained in Tim Pope article [A Note About Git Commit Messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+
+I tend to think the perfect is the enemy of the good. That is why I omitted talking about commit messages body. I think we have enough work to do on the summary line. But I would add one rule as counterpart:
+
+- Add a pull request/merge request tracking ID
+
+Most of the time, we use tools to discuss during a code review process like GitHub, Gitlab, ... All pieces of information about the change will be at least described there. I added this rule about tracking ID so it becomes easy to find related pull/merge request. Writing a commit body is better than adding tracking IDs because if you change your repository manager solution you will loose all this precious context information and your tracking IDs will be useless.
```

Comment:
```suggestion
Most of the time, we use tools to discuss during a code review process like GitHub, Gitlab, ... All pieces of information about the change will be at least described there. I added this rule about tracking ID so it becomes easy to find related pull/merge request. Writing a commit body is better than adding tracking IDs because if you change your repository manager solution, you will lose all this precious context information and your tracking IDs will be useless.
```